app/src/main/java/de/westnordost/streetcomplete/util/ktx/String.kt

```diff
@@ -4,3 +4,7 @@ fun String.truncate(length: Int): String =
     if (this.length > length) substring(0, length - 1) + "…" else this
 
 fun String.containsAll(words: List<String>) = words.all { this.contains(it) }
+
+/** Returns the indices of all occurrences of [substring]. */
+fun String.indicesOf(substring: String) =
+    generateSequence(indexOf(substring)) { indexOf(substring, it) }
```

Comment:
Uh, wait a moment... that works?

What about -1? If the substring cannot be found, the sequence should be empty, not consist of unlimited -1s.

And, shouldn't the offset be `it+1`?

Reply:
(using `sequence` with `yield` may result to be more correct and readable, maybe?)

Reply:
You're right, I didn't test it properly. I rewrote the implementation with `yield` and added a unit test: [`99aba20`](https://github.com/streetcomplete/StreetComplete/pull/5439/commits/99aba20c10349a4f1e16bc5cf99aa2a7eeec32bf)